src/utils/urlResolvers.ts

```diff
@@ -0,0 +1,67 @@
+import ChatSDKConfig from "../core/ChatSDKConfig";
+import LiveChatVersion from "../core/LiveChatVersion";
+import ChatAdapterProtocols from "../core/messaging/ChatAdapterProtocols";
+import libraries from "./libraries";
+
+const resolveIC3ClientUrl = (chatSDKConfig: ChatSDKConfig): string => {
+    if (chatSDKConfig.ic3Config && 'ic3ClientCDNUrl' in chatSDKConfig.ic3Config) {
+        return chatSDKConfig.ic3Config.ic3ClientCDNUrl as string;
+    }
+
+    if (chatSDKConfig.ic3Config && 'ic3ClientVersion' in chatSDKConfig.ic3Config) {
+        return libraries.getIC3ClientCDNUrl(chatSDKConfig.ic3Config.ic3ClientVersion);
+    }
+
+    return libraries.getIC3ClientCDNUrl();
+};
+
+const resolveChatAdapterUrl = (chatSDKConfig: ChatSDKConfig, liveChatVersion: LiveChatVersion, protocol: string): string => {

```

Comment:
could we break this factory in 3 different factories, one for each type of adapter, isolating the logic and validation in separate components,

Reply:
refactored